### PR TITLE
Immunization GET API

### PR DESCRIPTION
**Pull request by MonishaSivanathan**
- Changed all GET API as single API

### CODE REVIEW

File: routes/integration_pipeline_router.py [hunk 1]
```diff
@@ -53,33 +53,18 @@ async def hl7v2_vxu_v04(request: Request, raw_data: str =Body(..., media_type="t
...
+    immunization_id: Optional[str] = None,
     page: int = Query(1, ge=1), 
     count: int = Query(10, ge=1, le=100)
     ):
-    logger.info(f"Immunization for {patient_id} and vaccine for {name}")
-    return HL7ImmunizationClient.find_immunizations_by_patient_id(patient_id, name, page, count)
+    logger.info(f"Immunization for {patient_id} and vaccine for {vaccine_name}")
+    if vaccine_name:
+        return HL7ImmunizationClient.find_immunizations_by_patient_id(patient_id, vaccine_name, page, count)
+    elif patient_id and immunization_id:
```
Comment (tamsud) [inline routes/integration_pipeline_router.py:67]: patient_id check is necessary?

File: routes/integration_pipeline_router.py [hunk 2]
```diff
@@ -53,33 +53,18 @@ async def hl7v2_vxu_v04(request: Request, raw_data: str =Body(..., media_type="t
...
+        return HL7ImmunizationClient.get_immunizations_by_id(patient_id, immunization_id)
+    elif patient_id:
```
Comment (tamsud) [inline routes/integration_pipeline_router.py:69]: no need to check patient id as its default parameter

File: routes/integration_pipeline_router.py [hunk 3]
```diff
@@ -53,33 +53,18 @@ async def hl7v2_vxu_v04(request: Request, raw_data: str =Body(..., media_type="t
...
-async def get_immunization(request: Request, page: int = Query(1, ge=1), page_size: int = Query(10, ge=1, le=100), name: Optional[str] = None):
-    logger.info(f"Immunization name: {name}")
-    return HL7ImmunizationClient.get_immunization(name, page, page_size)
-
-
-@router.get("/immunization/vaccine_name/{patient_id}/{vaccine_name}")
-@permission_required("IMMUNIZATION", "READ")
 async def find_immunizations_by_patient_id(
     request: Request, 
     patient_id: Optional[str] = None,
```
Comment (tamsud) [inline routes/integration_pipeline_router.py:58]: can you remove optional its a path not a query param

**Review by tamsud — CHANGES_REQUESTED** [review]

File: routes/integration_pipeline_router.py (hunk 1 above)
Comment (MonishaSivanathan) [reply routes/integration_pipeline_router.py:67]: changed.

File: routes/integration_pipeline_router.py (hunk 2 above)
Comment (MonishaSivanathan) [reply routes/integration_pipeline_router.py:69]: sure, changed

File: routes/integration_pipeline_router.py (hunk 3 above)
Comment (MonishaSivanathan) [reply routes/integration_pipeline_router.py:58]: i have removed

**Review by tamsud — APPROVED** [review]

**Pull request merged**